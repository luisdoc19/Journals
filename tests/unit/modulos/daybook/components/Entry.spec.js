import { shallowMount } from "@vue/test-utils";
import EntryComponentVue from "@/modulos/daybook/components/EntryComponent.vue";
import { jornalState } from "../../../mock-data/test-jornal";

describe("Pruebas en Entry component", () => {
  const mockRouter = {
    push: jest.fn(),
  };

  const wrapper = shallowMount(EntryComponentVue, {
    props: {
      data: jornalState.entries[0],
    },
    global: {
      mocks: {
        $router: mockRouter,
      },
    },
  });

  it("Debe de hacer match con el snapshot", () => {
    expect(wrapper.html()).toMatchSnapshot();
  });

  it("debe redireccionar al hacer click en el entry-component", async () => {
    const div = wrapper.find(".entry-container");
    div.trigger("click");
    expect(mockRouter.push).toHaveBeenCalled();
  });
});
