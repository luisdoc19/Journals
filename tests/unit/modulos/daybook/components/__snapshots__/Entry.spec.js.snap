// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en Entry component Debe de hacer match con el snapshot 1`] = `
<div class="entry-container mb-3 pointer p-2">
  <div class="entry-title d-flex"><span class="text-success fs-5 fw-bold">05</span><span class="mx-1 fs-5">Aug</span><span class="mx-2 fw-light">2023</span></div>
  <div class="entry-description">No se que paso today</div>
</div>
`;
